Update game info generator

DanTheMan827 opened the pull request:
Updates the game info generator with https://github.com/MrKev312/AmiiboGameListGenerator/pull/6

AbandonedCart commented:
What about all of the local changes that were already made?

N3evin commented:
@DanTheMan827 does that solved the issue of rate limit? I had to make changes due to the rate limit on amiibo.life.

AbandonedCart commented:
> @DanTheMan827 does that solved the issue of rate limit? I had to make changes due to the rate limit on amiibo.life.

https://github.com/MrKev312/AmiiboGameListGenerator/pull/6/files#diff-943789b59693fa23b50a8f60a05cdedf4e52860acc95527c011939133c2c6672

DanTheMan827 commented:
@AbandonedCart what changes were already made? I went to the source repo, I didn’t know of any others.

As for 64-bit, does it really matter if it’s only running in a workflow? I updated to 8.0 because it’s LTS, and I also have a reusable workflow in the PR to @MrKev312 that could be used eventually.

@N3evin I noticed some issues with the server giving 500 errors, but when I removed the parallel for each, they seemed to have gone away… although it takes a bit longer to run.

AbandonedCart commented:
> @AbandonedCart what changes were already made? I went to the source repo, I didn’t know of any others.
> 
> As for 64-bit, does it really matter if it’s only running in a workflow? I updated to 8.0 because it’s LTS, and I also have a reusable workflow in the PR to @MrKev312 that could be used eventually.

I was referring to the changes made **locally** (as in here), such as addressing the rate limit. If you look in this repo, the binary has been updated multiple times despite the upstream source not being updated in quite some time.

If the aim is to use the current LTS, then you will need to update to 10 before the end of the year, as 8 actually becomes obsolete in November. Since it is a valid point that it only runs in a workflow here, I will gladly move that question to the original PR to be addressed separately.

N3evin commented:
> errors, but when I removed the parallel for each, they seemed to have gone away… although it takes a bit longer to run.

I won't mind longer to run, as of now i had to delay each call and also have to call it in a seperate batches to prevent rate limit blocking. If you run all of it at one go it will crash. If there's no real security concern I will not update as of right now, since we are just pulling data.

The update is done locally in my pc because it does not affect the original repo only for us when pulling the work flow.

As of now, is 5 amiibo per 15,000ms wait time to prevent hitting rate limit.

AbandonedCart commented:
Why doesn’t this check status codes instead of spamming retries for a blanket failure?

N3evin commented:
> @N3evin If you wouldn’t mind, why not post your changes to the generator so @DanTheMan827 can attempt to improve upon them? For whatever reason, he feels there is a problem with it, but is working with obsolete code. At least that way he won’t be trying to blindly recreate fixes you already included.

I had provided him the code. But he should be merging to gameinfo own repo instead of my code. I just changed the thread count nothing more and this merge will not be approved since it's directly merging the bin.

DanTheMan827 commented:
@N3evin When/if the other PR is merged, this one should be modified to use the reusable workflow provided by that repo.

N3evin commented:
> @N3evin When/if the other PR is merged, this one should be modified to use the reusable workflow provided by that repo.

The one with game_info.yml? Ok. Will look at how to trigger it and copy the compute game_info.json into the database folder 🤔
